### ISSUE TITLE
NoMethodError: undefined method `unpack1' for nil:NilClass

### ISSUE THREAD

**Issue by Baaboud** [rootstrap/active-storage-base64#76]
we get this error when there is an empty file is uploaded in our application because the data part is blank
for example `"data":"data:text/plain;base64,"`

sometimes we are expecting empty files to be uploaded

I think we should consider allowing empty data part to be blank or adding validation to allow or prevent empty files

```
lib/active_storage_support/base64_attach.rb:22
decoded_data  = Base64.decode64(data || '')
```

**Comment by santib** [comment]
Hi @Baaboud thanks for the report. Do you want to open a PR with the fix?

**Comment by Baaboud** [comment]
yes @santib I opened this [PR](https://github.com/rootstrap/active-storage-base64/pull/77) for this issue
please check it when you have time

**Comment by santib** [comment]
Fixed by #77

**Comment by santib** [comment]
@Baaboud done https://rubygems.org/gems/active_storage_base64/versions/3.0.1 sorry for the delay 😄
